react/features/chat/middleware.js

```diff
@@ -63,14 +68,17 @@ MiddlewareRegistry.register(store => next => action => {
             APP.API.notifyChatUpdated(unreadCount, isOpen);
         }
         break;
+
     case TOGGLE_CHAT:
-        unreadCount = 0;
-        isOpen = !getState()['features/chat'].isOpen;
+        isOpen = getState()['features/chat'].isOpen;
 
-        if (typeof APP !== 'undefined') {
-            APP.API.notifyChatUpdated(unreadCount, isOpen);
+        if (isOpen) {
```

Comment:
Can we move this logic to the action instead? When you do an async action you can check right there, instead of having to go through the middleware. Thsi way we'd get rid of the TOGGLE_CHAT action type.